django/chat/responses.py

```diff
@@ -417,6 +417,15 @@ async def add_files_to_library():
         input = response_message.parent.text
         source_nodes = retriever.retrieve(input)
 
+        if chat.options.qa_source_order == "reading_order":
+            source_nodes = sorted(
+                source_nodes,
+                key=lambda x: (
+                    x.node.relationships["1"].node_id,
```

Comment:
The metadata doesn't contain the document UUID - only the UUID for the data source. I can also envision a situation where multiple docs have the same title or URL. Thus, this seems like the best way to order by Document first (although they won't be alphabetized, of course)

Reply:
The metadata does include the document UUID - `node["metadata"]["doc_id"]`. You can retrieve the document itself from that, if you wanted to, i.e. `Document.objects.get(uuid_hex=node["metadata"]["doc_id"])`.

Could you please add a comment in the code about what `node.relationships["1"]` is? (If the doc_id change doesn't get rid of it)

Reply:
Just checked - there's only a `data_source_uuid` attribute in the metadata:
![image](https://github.com/user-attachments/assets/369c866d-6ed2-4c50-b149-402c3cd97970)


Comment added!

Reply:
Using ref_doc_id now, as discussed. See new sorting logic